src/prefect/contrib/tasks/databricks/databricks.py

```diff
@@ -0,0 +1,260 @@
+        if new_cluster is not None:
+            self._json["new_cluster"] = new_cluster
+        if libraries is not None:
+            self._json["libraries"] = libraries
+        if timeout_seconds is not None:
+            self._json["timeout_seconds"] = timeout_seconds
+        if self.name is not None:
+            self._json["run_name"] = self.name
+        if existing_cluster_id is not None:
+            self._json["existing_cluster_id"] = existing_cluster_id
+            self.logger.info(f"Using cluster ID: {self._json['existing_cluster_id']}")
+        else:
+            if existing_cluster_name is not None:
+                self._json["existing_cluster_id"] = self._get_cluster_id(
+                    existing_cluster_name
+                )
+                self.logger.info(
+                    f"Using cluster ID: {self._json['existing_cluster_id']}"
+                )
+
+    @property
+    def databricks_host(self) -> Optional[str]:
+        return self._databricks_host
+
+    @property
+    def timeout_seconds(self) -> int:
+        return self._timeout_seconds
+
+    @property
+    def databricks_retry_limit(self) -> int:
+        return self._databricks_retry_limit
+
+    @property
+    def databricks_retry_delay(self) -> int:
+        return self._databricks_retry_delay
+
+    @property
+    def polling_period(self) -> int:
+        return self._polling_period
+
+    @property
+    def polling_timeout(self) -> int:
+        return self._polling_timeout
+
+    @property
+    def json(self) -> dict:
+        return self._json
+
+    def _get_cluster_id(self, cluster_name: str) -> str:
+        """
+        Utility function to get a cluster id from cluster name
+        Args:
+            - name: cluster name
+
+        Returns:
+            - The cluster id
+        """
+        self.logger.info("Fetching existing clusters information ...")
+        query = f"{self._end_point}/clusters/list"
+        response = requests.get(
+            url=query,
+            timeout=self._timeout_seconds,
+            headers={
+                "Authorization": f"Bearer {self._databricks_token}",
+                "User-Agent": USER_AGENT,
+            },
+        )
+        response.raise_for_status()
+        data = response.json()
+        clusters = dict(
+            map(
+                lambda k: (k["default_tags"]["ClusterName"], k["cluster_id"]),
+                data["clusters"],
+            )
+        )
+
+        if cluster_name not in clusters:
+            raise Exception(f"Cluster {cluster_name} does not exist !")
+        return clusters[cluster_name]
+
+    def _wait_for_run_to_complete(
+        self, run_id: str, polling_period: int, timeout: int = 3600
+    ) -> dict:
+        """
+        Fetch Databricks run status via api/../jobs/run/get?run_id=<RUN_ID> until run state
+        is terminated. A run is terminated when matching either SUCCESS, FAILED, TIMEDOUT or CANCELED
```

Comment:
I don't believe this is is what the code does.  It doesn't check that the run state is is in the set you give, but rather considers the run terminated when any run state is given.